Sources/Rownd/Views/HubWebView/HubWebViewController.swift

```diff
@@ -386,6 +386,23 @@ extension HubWebViewController: WKScriptMessageHandler, WKNavigationDelegate {
                 break
             case .unknown:
                 break
+            case .authChallengeInitiated:
+                guard case .authChallengeInitiated(let authChallengeMessage) = hubMessage.payload else { return }
+                DispatchQueue.main.async {
+                    Context.currentContext.store.dispatch(SetAuthState(payload: AuthState(
+                        challengeId: authChallengeMessage.challengeId,
+                        userIdentifier: authChallengeMessage.userIdentifier
+                    )))
+                }
+                break
+            case .authChallengeCleared:
+                DispatchQueue.main.async {
+                    Context.currentContext.store.dispatch(SetAuthState(payload: AuthState(
```

Comment:
I think we need to make sure that other properties from AuthState are copied through here. (e.g., isVerifiedUser, hasPreviouslySignedIn, and userId)